tests/resolve_imports/file_parser.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gtest/gtest.h"
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <libcellml>
+#include <sstream>
+#include <vector>
+
+// generated with test resource locations
+#include "test_resources.h"
+
+static int countImportedComponents(libcellml::ModelPtr model)
```

Comment:
Is this something that you are thinking of supplying as a convenience function on the class at some point?  I think you can also use the iterate over items and test for isImport method?

Reply:
not sure yet - I'm just trying out things to see how it works...once I have something working I'll talk to you about how it should really be done ;-)